### ISSUE TITLE
Add Printexc.default_uncaught_exception_handler

### ISSUE THREAD

**Issue by dbuenzli** [ocaml/ocaml#9248]
That has the default uncaught exception handler. 

My use case is that I would like to treat specially uncaught exceptions in the initialisation code of a module but then otherwise not alter the default runtime system behaviour. 

Something like: 

```ocaml
let () = Printexc.set_caught_exception_handler myhandler
...
let () =
  let default = Printexc.default_uncaught_exception_handler in 
  Printexc.set_uncaught_exception_handler default
```  

(It may seem strange but in my scenario the source declarations `...` are not aware they are being surrounded by these two lets).

It's an easy task. The option in [this reference](https://github.com/ocaml/ocaml/blob/trunk/stdlib/printexc.ml#L274) should likely be evicted and initialised by a function named `default_uncaught_exception_handler` that has [this code](https://github.com/ocaml/ocaml/blob/trunk/stdlib/printexc.ml#L299-L301).

**Comment by gadmm** [comment]
The proposition looks sensible to me. Would you like to submit a PR? I can do the review.

(I am not sure about the junior-job label: if the goal of the label is to hook developers to become compiler hackers, ideally it should a non-boring task. I think a good junior-job task is interesting and simple enough that you had to force yourself not to code it over the afternoon.)

**Comment by gasche** [comment]
Junior jobs should not be tedious, but I don't think implementing the present PR would be "boring" in that sense. I think it is fine if the task is very easy, because for beginners the main source of difficulty is often the context rather than the contribution itself: cloning and building the compiler distribution, using the right git workflow, finding out what/where to change, and deadling with the PR submission and review process.

**Comment by dbuenzli** [comment]
Since we are on the meta of "junior jobs" I'd like to one last time advocate to call them *newcomer* jobs rather than *junior* jobs. The [definitions](https://www.merriam-webster.com/dictionary/junior) for junior may either not apply or I find them too elitistic or dismisive. Same (to a lesser extent tough) for "beginners".

@gadmm thanks for the review proposal, no time® for a PR right now hence the issue filing.

**Comment by gasche** [comment]
> Since we are on the meta of "junior jobs" I'd like to one last time advocate to call them newcomer jobs rather than junior jobs.

Done. (This was harder to change on Mantis.)

**Comment by sonologico** [comment]
This would actually be beneficial for one of my projects and I was looking for a good entry point to contributing to the compiler code base. I will make a PR for this if no one has started working on it already.

**Comment by gadmm** [comment]
Feel free to go ahead.

(I was speaking of my own experience of looking at "junior" jobs, but it seems you were right @gasche :)

**Comment by Armael** [comment]
Fixed by #9263.
